### ISSUE TITLE
y and y_pred shape error in _BaseClassification._check_shape()

### ISSUE THREAD

**Issue by szhang104** [pytorch/ignite#555]
Hello,

In a single-label, multiple-class classification task, suppose my model gives the output logits `y_pred` of shape (batch, num_classes, 1) and the correct results are `y` of shape (batch, 1), which are the integer labels, then the _check_shape() gives an error
`y must have shape of (batch_size, ...) and y_pred must have "
                             "shape of (batch_size, num_categories, ...) or (batch_size, ...), "
                             "but given {} vs {}.`

which happens because the code squeezes the second dimension of `y`. See https://github.com/pytorch/ignite/blob/c92838c950699839c37f57bb50f79f84ecb71fc3/ignite/metrics/accuracy.py#L24

It seems like the shapes are legal according to the message. Am I doing something wrong here, or what is the intended usage?

**Comment by vfdev-5** [comment]
@szhang104 thanks for the feedback! Yes, your input shapes seems to be not correctly covered by `Accuracy`.
A solution can be to provide `output_transform` function to get rid of these extra dimensions of size 1:
```python
def output_transform(output):
    y_pred, y = output
    return (y_pred[:, :, 0], y[:, 0])

acc = Accuracy(output_transform=output_transform)
```

HTH

**Comment by szhang104** [comment]
Thanks for the comment.
Is there a plan to change the squeezing logic to accommodate user input shapes like reported?

**Comment by vfdev-5** [comment]
@szhang104 yes, I think we need to review this logic
